pandas/tests/window/test_rolling.py

```diff
@@ -361,3 +361,60 @@ def test_rolling_datetime(self, axis_frame, tz_naive_fixture):
             }
         )
         tm.assert_frame_equal(result, expected)
+
+
+def test_rolling_window_as_string():
+    # see gh-22590
+    date_today = datetime.now()
+    days = pd.date_range(date_today, date_today + timedelta(365), freq="D")
+
+    np.random.seed(seed=421)
+
+    data = np.random.randint(1, high=100, size=len(days))
```

Comment:
can you use RandomState here instead so it doesn't globally set

Reply:
Done